src/main/java/io/swagger/template/TemplateApi.java

```diff
@@ -13,114 +13,197 @@
-
-                        proxy=proxy.replaceAll("path","\""+((RequestMapping)classNameInterfaces[0].getMethods()[i].getAnnotations()[2]).value()[0]+"\"");
-                        i++;
-                        proxy=proxy.replaceAll("parameter","@RequestParam(\""+((RequestParam)method.getParameters()[0].getAnnotations()[2]).value()+"\")"+method.getParameterTypes()[0].getName()+" "+((RequestParam) method.getParameters()[0].getAnnotations()[2]).value());
-                    } else {
-                        integrate = integrate.replaceAll("datatype", "");
-                        integrate = integrate.replaceAll("integrationspecific", "return new ResponseEntity<String>(" + ((String) iterator.get(integrationName)).toLowerCase() + "Proxy." + method.getName() + "(), HttpStatus.OK);");
-
-                        proxy=proxy.replaceAll("path","\""+((RequestMapping)classNameInterfaces[0].getMethods()[i].getAnnotations()[2]).value()[0]+"\"");
-                        i++;
-                        proxy=proxy.replaceAll("parameter","");
-                    }
-                    stringBuilderIntegrate.append(integrate);
-                    stringBuilderIntegrate.append("\n");
-
-
-                    stringBuilderProxy.append(proxy);
-                    stringBuilderProxy.append("\n");
-                } else {
-                    break;
-                }
+  public static void main(String[] args) throws Exception {
+
+    IntegrationName integrationName = new IntegrationName();
+    Field[] fieldList = integrationName.getClass().getFields();
+
+    for (Field iterator : fieldList) {
+
+      //Script to generate  CONTROLLER CODE(IntegrationNameAPI.java).
+      String content = IOUtils.toString(new FileReader(new File("TemplateName.java")));
+
+      FileWriter fileWriter =
+          new FileWriter(new File(iterator.get(integrationName) + "ApiController.java"));
+
+      String integration = IOUtils.toString(new FileReader(new File("IntegrationData.txt")));
+
+      String integrateString = IOUtils.toString(new FileReader(new File(
+          "/Users/prateekkoul/Downloads/spring-server-generated (1)/src/main/java/io/swagger/api/"
+              + iterator.get(integrationName) + "Api.java")));
+
+      String parameterData = IOUtils.toString(new FileReader(new File("ParameterData.txt")));
+
+      String functionParameterData =
+          IOUtils.toString(new FileReader(new File("FunctionParameterData.txt")));
+
+      String proxyParameterData =
+          IOUtils.toString(new FileReader(new File("ProxyParameterData.txt")));
+
+      content = content.replaceAll("Name", (String) iterator.get(integrationName));
+      content = content.replaceAll("Proxy", iterator.get(integrationName) + "Proxy");
+      content = content.replaceAll("proxyname",
+          ((String) iterator.get(integrationName)).toLowerCase() + "Proxy;\n");
```

Comment:
using a simple string search/replace is a reasonable start for do-it-yourself template.  However, there are many ready built templating engines that no only make it easier to do search/replace, but also allow you to add conditionals and looping logic directly in the template itself.  Might be worth trying something like Groovy's SimpleTemplateEngine or even velocity templates.

As an example of something a templating engine will handle gracefully - what if you want to have an attribute in your template class called `apiName`.  With line 52 above, the attribute will get changed to `api<IntegrationName>` which might not be what you actually want.  With a templating engine, you can typically specify what parts of the text should be taken literally, and which parts should be replaced via a `{{ paramName }}` or maybe `${paramName}` type syntax based on which templating engine you choose.

Reply:
Before starting with this script I tried to look into the **Velocity** framework, but got really perplexed at that time. I will start to understand **Velocity** framework or **Groovy** from today again and see how we can use that for our purpose.